c10/core/TensorImpl.cpp

```diff
@@ -430,74 +514,89 @@ bool _compute_non_overlapping_and_dense(
   return true;
 }
 
-bool TensorImpl::compute_non_overlapping_and_dense(identity<bool>) const {
-  if (is_sparse()) {
-    return false;
-  }
-  return _compute_non_overlapping_and_dense<int64_t>(
-      sizes_and_strides_.sizes_arrayref(),
-      sizes_and_strides_.strides_arrayref());
-}
-
 SymBool TensorImpl::compute_non_overlapping_and_dense(identity<SymBool>) const {
   if (is_sparse()) {
     return false;
   }
-  return _compute_non_overlapping_and_dense<c10::SymInt>(
-      extra_meta_->sizes_, extra_meta_->strides_);
+  SymIntArrayRef sizes = extra_meta_->sizes_;
+  SymIntArrayRef strides = extra_meta_->strides_;
+  // TODO: factor this logic somewhere nice
+  // Look for a SymNode to dispatch on
+  SymNode base;
+  for (const auto& s : sizes) {
+    if (s.is_symbolic()) {
+      base = s.toSymNodeImpl();
+      break;
+    }
+  }
+  if (!base) {
+    for (const auto& s : strides) {
+      if (s.is_symbolic()) {
+        base = s.toSymNodeImpl();
+        break;
+      }
+    }
+  }
+  if (!base) {
+    // Do the normal computation
+    return compute_non_overlapping_and_dense(identity<bool>());
```

Comment:
This is actually wrong 🤔  All the metadata is on `extra_meta_` but this is pulled from the wrong place